includes/wc-template-functions.php

```diff
@@ -1473,7 +1473,16 @@ function woocommerce_pagination() {
 	 * Output the product image before the single product summary.
 	 */
 	function woocommerce_show_product_images() {
-		wc_get_template( 'single-product/product-image.php' );
+		global $product;
+		$post_thumbnail_id = $product->get_image_id();
+		if ( ! $post_thumbnail_id ) {
+			$gallery_image_ids = $product->get_gallery_image_ids();
+			if ( ! empty( $gallery_image_ids ) ) {
+				$post_thumbnail_id = array_shift( $gallery_image_ids );
+			}
+		}
+		$args = compact( 'post_thumbnail_id', 'gallery_image_ids' );
```

Comment:
We don't use `$gallery_image_ids` in `single-product/product-image.php`, also this throws an "undefined variable" notice, since the  `$gallery_image_ids` may not exists.

```suggestion
```

Reply:
done in `e49527c`